xtraplatform-features/src/main/java/de/ii/xtraplatform/features/domain/FeaturePropertyV2.java

```diff
@@ -90,6 +92,30 @@ default Type getType() {
         }
     }
 
+    @Value.Immutable
+    @Value.Style(deepImmutablesDetection = true, builder = "new", attributeBuilderDetection = true)
+    @JsonDeserialize(builder = ImmutableConstraints.Builder.class)
+    interface Constraints {
+
+        Optional<String> getCodelist();
+
+        @JsonProperty(value = "enum")
+        Optional<List<String>> getEnumValues();
```

Comment:
`List` and `Map` don't have to be wrapped in `Optional`, they are already optional.